library/psa_its_file.c

```diff
@@ -51,6 +51,7 @@
 #define PSA_ITS_STORAGE_SUFFIX ".psa_its"
 #define PSA_ITS_STORAGE_FILENAME_LENGTH         \
     ( sizeof( PSA_ITS_STORAGE_PREFIX ) - 1 + /*prefix without terminating 0*/ \
+      16 + /*UID (64-bit number in hex)*/                               \
```

Comment:
This was caught by one of the printf warning flags. Basically the buffer is not big enough - the format specifier used is PSA_ITS_STORAGE_FILENAME_PATTERN or "%08lx%08lx". Each of those %08lx 's could product a 16 byte hex number, which would overflow the buffer.
The storage format has not changed, but this bug has potentially been with us for some time.

Reply:
Each `%08lx` prints a 32-bit number, so won't print more than 8 characters, making the total 16 characters for a 64-bit number. The 64-bit number is printed in two parts (upper then lower 32 bits) so that the code works on 32-bit platforms.

Reply:
Ah. This is why I already had to tame down that warning. In fairness to it, the input fed into it is a 64 bit number each time, however it doesn't take notice the shifted / masked part. Its a shame, in that this warning could be rather useful, however it tends to catch false positives a lot.

Reply:
Does it help to change `uid >> 32` to `( uid >> 32 ) & 0xffffffff`?

Reply:
Alternatively, since we assume that `int` is ≥32 bits anyway, it's ok to change the formats to `%08x` and cast the arguments to `unsigned` instead of `unsigned long`.